### PR TITLE
Added attach protocol field in volume attachment schema.

### PR DESCRIPTION
**Pull request by chitranm**
US60196: Added attach protocol field in volume attachment schema so that it can be used in decision making in host resources.

### CODE REVIEW

File: v1/api/swagger/components/schemas/VaAttachProtocol.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: >-
+  Current volume attachment protocol expressed as a string enum value (iscsi, nvmetcp, fuse, fc)
```
Comment (mchuang3) [inline v1/api/swagger/components/schemas/VaAttachProtocol.yaml:2]: Hi @chitranm, have you looked at ProtocolParameters.yaml and ProtocolKind.yaml in the schemas directory? These seems to be the old way of specifying attach protocol? See NewVolumeAttachment.yaml. These all needs to be consistent.

Comment (chitranm) [reply v1/api/swagger/components/schemas/VaAttachProtocol.yaml:2]: Hi @mchuang3, just now I was looking at quake code and found that Protocol related files (in hpegl-metal-client) are already present. I will revert this and modify them. 
Thanks.

File: v1/api/swagger/components/schemas/FCParameters.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+type: object
+required:
+  - HostIPAddress
+  - InitiatorName
+properties:
+  HostIPAddress:
+    type: string
+    description: The IP address of the host in dotted notation.
+    example: 10.1.2.3
```
Comment (shi98382) [inline v1/api/swagger/components/schemas/FCParameters.yaml:9]: I don't see a need for this field. you should have IP in host.

File: v1/api/swagger/components/schemas/FCParameters.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  InitiatorName:
+    type: string
+    description: >-
+      FC Initiator world wide port names.
```
Comment (shi98382) [inline v1/api/swagger/components/schemas/FCParameters.yaml:13]: This is confusing. You either specify this field as a FC initiator or HBA wwpn.

File: v1/api/swagger/components/schemas/ProtocolParameters.yaml [hunk 4]
```diff
@@ -7,3 +7,8 @@ properties:
     description: >-
       The ISCSI parameters for this attachment. This value must be specified if
       Protocol is iscsi
+  FC:
+    $ref: ./FCParameters.yaml
+    description: >-
+      The FC parameters for this attachment. This value must be specified if
+      Protocol is fc
```
Comment (shi98382) [inline v1/api/swagger/components/schemas/ProtocolParameters.yaml:14]: Be consistent with term. I would make a global change on "fc" with "FC"

File: v1/pkg/client/api/openapi.yaml [hunk 5]
```diff
@@ -3354,6 +3354,9 @@ components:
             CHAPSecret: CHAPSecret
             CHAPUserName: CHAPUserName
             InitiatorName: iqn.1234.com.hpe-initiator
+          FC:
+            HostIPAddress: 10.1.2.3
+            InitiatorName: 51:40:2e:c0:17:32:9a:e8
```
Comment (shi98382) [inline v1/pkg/client/api/openapi.yaml:3359]: same here.

File: v1/pkg/client/api/openapi.yaml [hunk 6]
```diff
@@ -4771,11 +4774,16 @@ components:
           CHAPSecret: CHAPSecret
           CHAPUserName: CHAPUserName
           InitiatorName: iqn.1234.com.hpe-initiator
+        FC:
+          HostIPAddress: 10.1.2.3
+          InitiatorName: 51:40:2e:c0:17:32:9a:e8
```
Comment (shi98382) [inline v1/pkg/client/api/openapi.yaml:4779]: same here.

File: v1/api/swagger/components/schemas/FCParameters.yaml (hunk 3 above)
Comment (chitranm) [reply v1/api/swagger/components/schemas/FCParameters.yaml:13]: Ok, I will keep FC initiator name.

File: v1/api/swagger/components/schemas/ProtocolParameters.yaml (hunk 4 above)
Comment (chitranm) [reply v1/api/swagger/components/schemas/ProtocolParameters.yaml:14]: Sure will keep FC.

File: v1/pkg/client/api/openapi.yaml (hunk 6 above)
Comment (chitranm) [reply v1/pkg/client/api/openapi.yaml:4779]: If this comment is for HostIPAddress is not needed - please check https://github.com/hpe-hcss/quake/blob/1638e3643ed0be37a15e3148003bd8036fa1ef44/portal/pkg/volumeattachment/translate.go#L54.

File: v1/api/swagger/components/schemas/FCParameters.yaml (hunk 3 above)
Comment (shi98382) [reply v1/api/swagger/components/schemas/FCParameters.yaml]: I think about it. The current backend can provide in term of attachment is the host ID and volume ID, nothing more. That's something that you can only mapping over to project API. The description of this field needs to revised to reflect the nature to volume attachment.

**Comment by shi98382** [comment]
LGTM.

**Comment by chitranm** [comment]
Hi @mchuang3, if you have no concerns shall I merge this PR?

**Review by mchuang3 — APPROVED** [review]

**Comment by mchuang3** [comment]
> Hi @mchuang3, if you have no concerns shall I merge this PR?

Looks good. Go for it. Thanks.

**Pull request merged**